zcash_keys/src/encoding.rs

```diff
@@ -76,13 +76,24 @@ where
     }
 }
 
+/// A trait for encoding and decoding Zcash addresses.
 pub trait AddressCodec<P>
```

Comment:
I'm happy to merge the PR with these docs, but this trait is almost certainly going away (as the address encoding and decoding logic here is duplicative of `zcash_address`).